### PR TITLE
Fixing FlowFixme errors

### PR DESCRIPTION
**Pull request by mitermayer**
This PR removes the $FlowFixMe and fixes the flow errors as well as minor things uncovered by the flow parser.

### CODE REVIEW

File: src/component/base/DraftEditorPlaceholder.react.js [hunk 1]
```diff
@@ -25,7 +24,6 @@ type Props = {
   accessibilityID: string,
   editorState: EditorState,
   text: string,
-  textAlignment: DraftTextAlignment,
```
Comment (mitermayer) [inline src/component/base/DraftEditorPlaceholder.react.js]: We never used the alignment property in this component. Is that a feature that folks would need ?

File: src/component/base/DraftEditor.react.js [hunk 2]
```diff
@@ -300,10 +295,6 @@ class DraftEditor extends React.Component<DraftEditorProps, State> {
             tabIndex={this.props.tabIndex}>
             {}
```
Comment (mitermayer) [inline src/component/base/DraftEditor.react.js]: This rogue '{}' is fixed on my other PR

File: src/component/base/DraftEditorPlaceholder.react.js (hunk 1 above)
Comment (flarnie) [reply src/component/base/DraftEditorPlaceholder.react.js]: Folks in open source may be using it - if we are deprecating we should warn through the 0.11.* series and then remove in 0.12.*, similar to how React handles deprecations.

File: src/component/base/DraftEditor.react.js [hunk 3]
```diff
@@ -202,14 +202,9 @@ class DraftEditor extends React.Component<DraftEditorProps, State> {
...
     if (this._showPlaceholder()) {
       return (
-        /* $FlowFixMe(>=0.53.0 site=www,mobile) This comment suppresses an
-         * error when upgrading Flow's support for React. Common errors found
-         * when upgrading Flow's React support are documented at
-         * https://fburl.com/eq7bs81w */
         <DraftEditorPlaceholder
           text={nullthrows(this.props.placeholder)}
           editorState={this.props.editorState}
-          textAlignment={this.props.textAlignment}
```
Comment (flarnie) [inline src/component/base/DraftEditor.react.js]: I think we still want to do this with `textAlignment` - and I did see uses of `textAlignment` internally at FB.

File: src/component/base/DraftEditorPlaceholder.react.js (hunk 1 above)
Comment (flarnie) [reply src/component/base/DraftEditorPlaceholder.react.js]: We did - see your previous fix. It should match the `textAlignment` passed in as a prop to `DraftEditor`.

File: src/component/base/DraftEditorProps.js [hunk 4]
```diff
@@ -60,10 +60,10 @@ export type DraftEditorProps = {
   // For a given `ContentBlock` object, return an object that specifies
   // a custom block component and/or props. If no object is returned,
   // the default `DraftEditorBlock` is used.
-  blockRendererFn?: (block: ContentBlock) => ?Object,
+  blockRendererFn: (block: ContentBlock) => ?Object,
```
Comment (flarnie) [inline src/component/base/DraftEditorProps.js]: What if a user doesn't pass this into Draft? And same question about the props below.

File: src/component/contents/DraftEditorBlock.react.js [hunk 5]
```diff
@@ -43,15 +44,15 @@ type Props = {
   contentState: ContentState,
   block: ContentBlock,
   customStyleMap: Object,
-  customStyleFn: Function,
+  customStyleFn: (style: DraftInlineStyle, block: ContentBlock) => ?Object,
```
Comment (flarnie) [inline src/component/contents/DraftEditorBlock.react.js]: :D Love the specific typing here.

File: src/component/contents/DraftEditorBlock.react.js [hunk 6]
```diff
@@ -43,15 +44,15 @@ type Props = {
...
   tree: List<any>,
   selection: SelectionState,
-  decorator: DraftDecoratorType,
+  decorator: ?DraftDecoratorType,
   forceSelection: boolean,
   direction: BidiDirection,
   blockProps?: Object,
   startIndent?: boolean,
-  blockStyleFn: Function,
```
Comment (flarnie) [inline src/component/contents/DraftEditorBlock.react.js]: Nice catch - looks like we now handle this in a parent component.

File: src/component/contents/DraftEditorLeaf.react.js [hunk 7]
```diff
@@ -45,7 +45,7 @@ type Props = {
 
   // The current `SelectionState`, used to represent a selection range in the
   // editor
-  selection: SelectionState,
+  selection: ?SelectionState,
```
Comment (flarnie) [inline src/component/contents/DraftEditorLeaf.react.js]: Would it work to do `SelectionState | null`?

**Comment by flarnie** [comment]
PS Thanks so much for this clean-up! Great to see this crufty Flow error stuff being washed away. 💦✨

File: src/component/base/DraftEditor.react.js (hunk 3 above)
Comment (mitermayer) [reply src/component/base/DraftEditor.react.js]: The reason why I remove was due to the fact that `DraftEditorPlaceholder` does not use the prop `textAlignment`.

If you refer to `https://github.com/facebook/draft-js/blob/master/src/component/base/DraftEditorPlaceholder.react.js` you will see no usage of `textAlignment` apart from the props definition itself

But if you prefer I can updated this PR to still have to pass `textAlignment` and make it a nullable `?textAlignment` instead.

File: src/component/base/DraftEditorProps.js (hunk 4 above)
Comment (mitermayer) [reply src/component/base/DraftEditorProps.js]: The reason why we were able to delete it was that we already supply a default empty function for it https://github.com/facebook/draft-js/blob/master/src/component/base/DraftEditor.react.js#L73

File: src/component/contents/DraftEditorLeaf.react.js (hunk 7 above)
Comment (mitermayer) [reply src/component/contents/DraftEditorLeaf.react.js]: We could do that, I have no strong opinion around it, we are currently doing a non strict check for it on https://github.com/facebook/draft-js/blob/master/src/component/contents/DraftEditorLeaf.react.js#L86

**Comment by mitermayer** [comment]
Will import this internally to see if this flow type changes affect any internal project and do further amends from within phabricator

**Pull request closed without merging**